pyhuman/app/workflows/browse_web.py

```diff
@@ -31,15 +39,30 @@ def action(self, extra=None):
     """ PRIVATE """
 
     def _web_browse(self):
-        random_website = self._get_random_website()
-        try:
-            self.driver.driver.get('https://' + random_website)
-            sleep(self.input_wait_time)
-        except Exception as e:
-            print('Error loading random website %s: %s' % (random_website.rstrip(), e))
+        self.driver.driver.set_page_load_timeout(self.default_timeout)
+        self._browse(self._get_random_website())
+        sleep(self.default_wait_time)
+        self._navigate_website()
+
 
     def _get_random_website(self):
-        return random.choice(self.website_list)
+        # Get a random website from the list of websites
+        try:
+            return random.choice(self.website_list)
+        except Exception as error:
```

Comment:
Just a thought, what happens if an exception is thrown here?

Trying to quickly think through it, `_get_random_website` would return `None`, which then gets passed to `_browse`, which would then be forced to throw an exception due to invalid input or something along those lines. It wouldn't necessarily break things since it seems like you've been thorough with making sure everything that needs to be caught has a try-except, but it may be worth considering what are the cases where an exception may be thrown and how do you want the rest of the workflow to handle the fact that an earlier step is returning `None`. For example, would `random.choice` only throw an exception if `self.website_list` is empty? In that case you may be able to handle that edge case earlier in the code to avoid issues here.

Food for thought, not sure whether it even needs any changes. Just considering that there are edge cases where try-excepts can make things more difficult to debug, it is sometimes useful to examine the exact reason it's being used in a particular spot.

Reply:
Great catch, thanks for spotting that. I don't think it makes sense to catch the error here because if there's an issue retrieving the random website from the list, that needs to be dealt with